charts/generic-service/templates/_helpers.tpl

```diff
@@ -55,9 +55,45 @@ release: {{ .Release.Name }}
 {{- end }}
 
 {{/*
-Create a string from a list of values joined by a comma
+Create IP allow list annotation form nginx
 */}}
-{{- define "app.joinListWithComma" -}}
-{{- $local := dict "first" true -}}
-{{- range $k, $v := . -}}{{- if not $local.first -}},{{- end -}}{{- $v -}}{{- $_ := set $local "first" false -}}{{- end -}}
+{{- define "app.makeIpAllowList" -}}
+{{- $allCustomItemNames := list -}}
+{{- $allGroups := list -}}
+{{- $allGroupItemNames := list -}}
+{{- $allAllowlists := list -}}
+  {{- range $k, $v := .Values.allowlist -}}
+    {{- if and (ne $k "groups") (kindIs "string" $v) -}}
+      {{- $allAllowlists = append $allAllowlists $v -}}
+      {{- $allCustomItemNames = append $allCustomItemNames $k -}}
+    {{- end -}}
+    {{- if eq $k "groups" -}}
+      {{- $_ := required "'groups' key found in 'allowlist' values, so 'allowlist_groups' is a required value but was not supplied." $.Values.allowlist_groups }}
+      {{- $groups := $v -}}
+      {{- range $group := $groups -}}
+        {{- if not (get $.Values.allowlist_groups $group) -}}
+          {{- required "Invalid group found in 'allowlist', check defined lists in 'allowlist_groups'." "" }}
```

Comment:
not required now but as a polishing step, could potentially include list of available groups in log message

Reply:
yes, thought might add that once we know for sure what they are.